docs/examples/Tutorials/Tutorial4/DeviceGroupQuery.cs

```diff
@@ -116,7 +113,12 @@ protected override void OnReceive(object message)
                     Context.Become(WaitingForReplies(repliesSoFar, stillWaiting));
                 }
             }
-            #endregion
+            #endregion   
+
+            protected override void OnReceive(object message)
```

Comment:
We should probably just refactor this actor to do all of its actual message handling inside `OnReceive` instead of having this behavior-switching. We could just make the state private members. Although this is a cool way of showing how you can use closures to build receive methods, which is a valid technique in practice.

Reply:
I think the purpose of this tutorial was to teach `using the Context.Become() trick instead of just making the repliesSoFar and stillWaiting structures mutable fields of the actor`

Reply:
Got it. In that case, we should leave it as-is. Want to teach the concept.

Reply:
So, if we change strategy in this actor, we have to change the documentation too. I think this is a good way for the tutorial.